### PR TITLE
Added support for EIO404 Bluetooth Mesh IoT Base Station. (#78)

### PR DESCRIPTION
**Pull request by JeremyTheocharis**
* Added support for EIO404 Bluetooth Mesh IoT Base Station. Changed device identification to use URI instead of por

* Apply Suggestions from PR

* Apply some suggestions from coderabbitai

## Summary by CodeRabbit

- **New Features**
	- Enhanced `README.md` with detailed descriptions of plugins and configuration options.
	- Added new sections for Modbus, SensorConnect, and Beckhoff ADS plugins.
	- Improved clarity on OPC UA and S7comm plugins with expanded functionality and configuration details.
	- Introduced a new method to check Bluetooth mesh compatibility for devices.

- **Bug Fixes**
	- Streamlined handling of connected devices across various methods, improving error reporting and data retrieval.

- **Documentation**
	- Comprehensive updates to the `README.md` to aid developers in understanding plugin functionalities and configurations.
	- Updated test cases to reflect changes in the handling of connected devices and methods.
	- Added usage comments in the Makefile for better guidance on test execution.

### CODE REVIEW

**Comment by JeremyTheocharis** [comment]
@EmlingerM some tests are still failing because of the rename, but happy to fix them

```
Error: ./customIodd_test.go:42:4: unknown field CurrentPortMap in struct literal of type sensorconnect_plugin.SensorConnectInput
Error: ./downloadPortModeData_test.go:39:27: input.GetUsedPortsAndMode undefined (type *sensorconnect_plugin.SensorConnectInput has no field or method GetUsedPortsAndMode, but does have unexported method getUsedPortsAndMode)
Error: ./downloadSensorData_test.go:40:27: input.GetUsedPortsAndMode undefined (type *sensorconnect_plugin.SensorConnectInput has no field or method GetUsedPortsAndMode, but does have unexported method getUsedPortsAndMode)
Error: ./downloadSensorData_test.go:42:11: input.CurrentPortMap undefined (type *sensorconnect_plugin.SensorConnectInput has no field or method CurrentPortMap)
```

File: sensorconnect_plugin/sensorconnect.go [hunk 1]
```diff
@@ -198,33 +200,35 @@ func (s *SensorConnectInput) ReadBatch(ctx context.Context) (service.MessageBatc
...
-						info.ProductName,
-						info.Serial,
+						device.BtAdapter,
+						device.Port,
+						device.Mode,
+						device.Connected,
+						device.DeviceID,
+						device.VendorID,
+						device.ProductName,
+						device.Serial,
```
[the remark this answers is not in the document]
Comment (JeremyTheocharis) [reply sensorconnect_plugin/sensorconnect.go:231]: I don't understand this. why do we need to have a lock for s.GetConnectedDevices(ctx)?

File: sensorconnect_plugin/downloadSensorData.go [hunk 2]
```diff
@@ -46,27 +46,27 @@ func (s *SensorConnectInput) GetSensorDataMap(ctx context.Context) (map[string]i
...
 			continue
 		}
 
 		var query string
-		switch info.Mode {
+		switch device.Mode {
 		// DI mode
 		case 1:
-			query = fmt.Sprintf("/iolinkmaster/port[%d]/pin2in", port)
+			query = device.Uri + "/pin2in"
```
[the remark this answers is not in the document]
Comment (JeremyTheocharis) [reply sensorconnect_plugin/downloadSensorData.go:61]: the risk of injection is just a number here, so it is low

File: sensorconnect_plugin/downloadPortModeData.go [hunk 3]
```diff
@@ -210,33 +240,64 @@ func (s *SensorConnectInput) getUsedPortsAndMode(ctx context.Context) (RawUsedPo
...
-	if err != nil {
-		return RawUsedPortsAndMode{}, err
-	}
+	// To avoid HTTP 413 "Payload Too Large" or 507 "Insufficient Storage" errors, we need to request data in batches, splitting the payload into smaller, manageable chunks.
+	result := map[string]UPAMDatum{}
+	var errstrings []string
+
+	for offset := 0; offset < len(dataPoints); offset += BadgeSize {
+		limit := min(BadgeSize, len(dataPoints)-offset)
+		requestData := map[string]interface{}{
```
[the remark this answers is not in the document]
Comment (JeremyTheocharis) [reply sensorconnect_plugin/downloadPortModeData.go:263]: this is a golang builtin code 

// The min built-in function returns the smallest value of a fixed number of
// arguments of [cmp.Ordered] types. There must be at least one argument.
// If T is a floating-point type and any of the arguments are NaNs,
// min will return NaN.
func min[T cmp.Ordered](x T, y ...T) T


so ignoring this

File: sensorconnect_plugin/sensorconnect.go [hunk 4]
```diff
@@ -155,34 +155,36 @@
...
-			info.ProductName,
-			info.Serial,
+			device.BtAdapter,
+			device.Port,
+			device.Mode,
+			device.Connected,
+			device.DeviceID,
+			device.VendorID,
+			device.ProductName,
+			device.Serial,
```
[the remark this answers is not in the document]
Comment (JeremyTheocharis) [reply sensorconnect_plugin/sensorconnect.go:187]: its not sensitive

**Review by kanapuli — APPROVED** [review]

**Pull request merged**